docs/spec/concepts.rst

```diff
@@ -3,6 +3,268 @@
+C++ are examples of statically-typed object-oriented languages.
+
+A **dynamically-typed** programming language does not run a type-checker before
+running a program. Instead, it checks the types of values before performing
+operations on them at runtime. This is not to say that the language is
+"untyped". Values at runtime have a type and their uses obey typing rules. Not
+every operation will be checked, but certain primitive operations in the
+language such as attribute access or arithmetic are. Python is a
+dynamically-typed language.
+
+**Gradual typing** is a way to combine static and dynamic typing.
+Type-annotated Python allows opting in to static type checking at a fine level
+of granularity, so that some type errors can be caught statically, before
+running the program. Variables, parameters, and returns can optionally be given
+a static type annotation. Even within the type of a single data structure,
+static type checking is opt-in and granular. For example, a dictionary can be
+annotated to have static checking of the key type but only dynamic checking of
+the value type.
+
+In gradual typing, a special "unknown" or "dynamic" type is used in the static
+type system for names or expressions whose type is not known statically.  In
+Python, this type is spelled :ref:`Any`. Because :ref:`Any` indicates a
+statically-unknown type, the static type checker can't check type correctness
+of operations on expressions typed as :ref:`Any`. These operations are
+dynamically checked, via the Python runtime's usual dynamic checking.
+
+The Python type system also uses ``...`` within :ref:`Callable` types and
+within ``tuple[Any, ...]`` (see :ref:`tuples`) to indicate a statically-unknown
+component of a type. The detailed rules for these usages are discussed in their
+respective sections of the specification.
+
+This specification describes a gradual type system for Python.
+
+Fully static and gradual types
+------------------------------
+
+We will refer to types that do not contain :ref:`Any` (or the ``...`` in
+``Callable`` or ``tuple[Any, ...]``) as a sub-part as **fully static types**.
+
+A **gradual type** can be a fully static type, :ref:`Any` itself, or a type
+that contains :ref:`Any` (or the ``...`` in ``Callable`` or ``tuple[Any,
+...]``) as a sub-part.
+
+Fully static types
+~~~~~~~~~~~~~~~~~~
+
+A fully static type denotes a set of potential runtime values. For instance,
+the fully static type ``object`` is the set of all Python objects. The fully
+static type ``bool`` is the set of values ``{ True, False }``. The fully static
+type ``str`` is the set of all Python strings; more precisely, the set of all
+Python objects whose runtime type (``__class__`` attribute) is either ``str``
+or a class that inherits directly or indirectly from ``str``. A :ref:`Protocol
+<Protocols>` denotes the set of all objects which share a certain set of
+attributes and/or methods.
```

Comment:
This is an interesting observation, but I think the distinction would be lost on most readers unless we explicitly highlighted it, and explicitly highlighting it would be an unnecessary digression here, I think. So I'm inclined not to extend this paragraph with an additional example.